### ISSUE TITLE
Craft setup fails when you have one or more linkit fields and uses project.yaml

### ISSUE THREAD

**Issue by nettum** [fruitstudios/craft-linkit#57]
To reproduce: 
1. `composer create-project craftcms/craft craft3_fresh`
2. `cd craft3_fresh`
3. Create database
4. `./craft setup` 
5. `composer require fruitstudios/linkit`
6. `./craft install/plugin linkit`
7. Create a linkit field
8. Drop database
9. Run `./craft setup` again
...
```
> applying existing project config ... done
Exception: No primary site exists (/home/nettum/Sites/temp/craft3_fresh/vendor/craftcms/cms/src/services/Sites.php:496)
...
Exception 'craft\errors\MigrationException' with message 'An error occurred while executing the "craft\migrations\Install migration: No primary site exists
```

This seems to happen because of `$projectConfig->muteEvents = true;` in the Install.php migration. 
If I set this to false or remove it all together things seems to work.

project.yaml:
```
dateModified: 1551862082
email:
  fromEmail: nettum@gmail.com
  fromName: 'craft3 fresh'
  transportType: craft\mail\transportadapters\Sendmail
fieldGroups:
  07fe101c-b8a2-4d7d-888a-a329255d0283:
    name: Common
fields:
  7162b828-f22d-490c-ac7f-bd8a28e63b40:
    contentColumnType: text
    fieldGroup: 07fe101c-b8a2-4d7d-888a-a329255d0283
    handle: myLink
    instructions: ''
    name: 'My Link'
    searchable: true
    settings:
      allowCustomText: ''
      allowTarget: ''
      defaultText: ''
      selectLinkText: ''
      types:
        fruitstudios\linkit\models\Asset:
          customLabel: ''
          customSelectionLabel: ''
          enabled: ''
          sources: '*'
        fruitstudios\linkit\models\Category:
          customLabel: ''
          customSelectionLabel: ''
          enabled: ''
          sources: '*'
        fruitstudios\linkit\models\Email:
          customLabel: ''
          customPlaceholder: ''
          enabled: ''
        fruitstudios\linkit\models\Entry:
          customLabel: ''
          customSelectionLabel: ''
          enabled: '1'
          sources: '*'
        fruitstudios\linkit\models\Facebook:
          customLabel: ''
          customPlaceholder: ''
          enabled: ''
        fruitstudios\linkit\models\Instagram:
          customLabel: ''
          customPlaceholder: ''
          enabled: ''
        fruitstudios\linkit\models\LinkedIn:
          customLabel: ''
          customPlaceholder: ''
          enabled: ''
        fruitstudios\linkit\models\Phone:
          customLabel: ''
          customPlaceholder: ''
          enabled: ''
        fruitstudios\linkit\models\Twitter:
          customLabel: ''
          customPlaceholder: ''
          enabled: ''
        fruitstudios\linkit\models\Url:
          allowAlias: '1'
          allowHash: '1'
          allowMailto: '1'
          allowPaths: '1'
          customLabel: ''
          customPlaceholder: ''
          enabled: '1'
        fruitstudios\linkit\models\User:
          customLabel: ''
          customSelectionLabel: ''
          enabled: ''
          sources: '*'
          userPath: ''
    translationKeyFormat: null
    translationMethod: none
    type: fruitstudios\linkit\fields\LinkitField
plugins:
  linkit:
    edition: standard
    enabled: true
    schemaVersion: 1.0.8
siteGroups:
  3a3339d0-b1f5-4243-a52f-a8d017eb5470:
    name: 'craft3 fresh'
sites:
  da69fd41-8b78-4ea7-a321-c2ffe849b42e:
    baseUrl: $DEFAULT_SITE_URL
    handle: default
    hasUrls: true
    language: en-US
    name: 'craft3 fresh'
    primary: true
    siteGroup: 3a3339d0-b1f5-4243-a52f-a8d017eb5470
    sortOrder: 1
system:
  edition: solo
  live: true
  name: 'craft3 fresh'
  schemaVersion: 3.1.25
  timeZone: America/Los_Angeles
users:
  allowPublicRegistration: false
  defaultGroup: null
  photoSubpath: ''
  photoVolumeUid: null
  requireEmailVerification: true
```

**Comment by nettum** [comment]
This seems to be fixed in [Craft 3.1.17](https://github.com/craftcms/cms/blob/develop/CHANGELOG-v3.md#3117---2019-03-08)

**Comment by nettum** [comment]
Reopen this. Will make an issue on craft cms with more details.

**Comment by samhibberd** [comment]
I have rewritten the migration again, hoping 1.1.10 addresses Linkit's side of things.

**Comment by nettum** [comment]
Thank you, will do some test migrations when 1.1.10 is out.

**Comment by nettum** [comment]
Still seems to silently fail when running migrations on `./craft install`.
If I set `$projectConfig->muteEvents = false;` line 36 in Install.php it works.
More details in https://github.com/craftcms/cms/issues/3977. Test-cases there are still valid.
It works if the migrations is run on admin pageload and not from cli, so I guess this is something the guys a pixel and tonic should look at as migration is not consistent depending on running from cli/admin. Most likely happens to more plugins than linkit.

**Comment by nettum** [comment]
For the record, this was fixed in v1.1.11
